### PR TITLE
FEAT:adding lesson06 assignment all tests passed

### PR DESCRIPTION
**Pull request by HeirJordan3**

### CODE REVIEW

File: lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java [hunk 1]
```diff
@@ -21,7 +21,12 @@ public static void main(String[] args) {
    * @return True if the age corresponds to a voting age and false otherwise.
    */
   public static boolean canVote(int age) {
-    return false;
+    if (age >= 18) {
```
Comment (anthonydmays) [inline lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java:24]: Would be cleaner as a tenary expression.

File: lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java [hunk 2]
```diff
@@ -80,8 +128,37 @@ public static double addNumbers(double[] values) {
    * @param n The first `n` of fibonacci values to compute.
    * @return An array containing the first `n` fibonacci values.
    */
+
+  // Validate Input: Ensure n is positive. Return an empty array for invalid input.
```
Comment (anthonydmays) [inline lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java:132]: These comments should go into the method and annotate the lines of code themselves. If you no longer need these, just remove.

File: lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java [hunk 3]
```diff
@@ -95,7 +172,7 @@ public static int[] getFirstNFibonacciNumbers(int n) {
    */
   public static int binarySearch(int[] values, int start, int end, int value) {
     if (end < start) {
-      // The range is not valid so just return -1.
```
Comment (anthonydmays) [inline lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java:98]: I'd rather keep this comment around :D

**Review by anthonydmays — CHANGES_REQUESTED** [review]

File: lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java [hunk 4]
```diff
@@ -80,8 +128,37 @@ public static double addNumbers(double[] values) {
...
+  // Handle Base Cases: Directly assign 1 to the first (and second if n > 1) elements.
+
+  // Calculate Fibonacci: Loop from the third element, each value is the sum of the two preceding
+  // ones.
+
+  // Return Array: Output the array with the first n Fibonacci numbers.
   public static int[] getFirstNFibonacciNumbers(int n) {
-    return null;
+
+    if (n < 1) {
```
Comment (anthonydmays) [inline lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java:144]: Try doing all of this in just a single `for` loop.

**Review by anthonydmays — CHANGES_REQUESTED** [review]

File: lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java (hunk 1 above)
Comment (anthonydmays) [reply lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java:24]: Misspoke, you don't even need a ternary. You can make this a one-line return statement though.

File: lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java [hunk 5]
```diff
@@ -61,7 +97,13 @@ public static String convertGpaToLetterGrade(double gpa) {
    * @return The factorial of n.
    */
   public static int computeFactorial(int n) {
-    return 0;
+    int factorial = 1;
+
+    for (int i = 1; i <= n; i++) {
```
Comment (anthonydmays) [inline lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java:102]: What if n is negative? This won't work.

**Review by anthonydmays — CHANGES_REQUESTED** [review]

File: lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java (hunk 5 above)
Comment (anthonydmays) [reply lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java:102]: More specifically, we should throw an exception if `n < 0`, but we haven't covered that yet. I'll allow this for now.

File: lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java (hunk 4 above)
Comment (HeirJordan3) [reply lesson_06/conditionals/conditionals_app/src/main/java/com/codedifferently/lesson6/Lesson6.java:144]: hmm interesting will do !

**Review by anthonydmays — APPROVED** [review]
LGTM

**Pull request closed without merging**